app/controllers/finance/ecf/participant_declarations/void_controller.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module Finance
+  module ECF
+    module ParticipantDeclarations
+      class VoidController < BaseController
+        before_action :set_participant_profile_and_declaration
+
+        def new; end
+
+        def create
+          VoidParticipantDeclaration.new(@declaration, voided_by_user: current_user).call
+          set_success_message(**I18n.t("finance.void_declaration.success"))
+        rescue Api::Errors::InvalidTransitionError
+          set_important_message(**I18n.t("finance.void_declaration.failure"))
+        ensure
+          redirect_to finance_participant_path(@participant_profile.user)
+        end
+
+      private
+
+        def set_participant_profile_and_declaration
+          @participant_profile = ParticipantProfile.find(params[:participant_profile_id])
```

Comment:
can't we infer the profile from the declaration? just wondering why we'll need to have both ids sent when we only need to the declaration?

Reply:
Yeah we could, but I opted to stay consistent with other similar routes here that include the participant profile ID. Happy to change it but we may want to do it as part of a wider refactoring to keep them consistent, for example:

```
/finance/participant_profiles/:participant_profile_id/ecf/induction_records/:induction_record_id/change_training_status/new
```

Reply:
makes sense, let's keep it